src/components/z-breadcrumb/index.tsx

```diff
@@ -0,0 +1,271 @@
+    this.homepageNode = this.pathsList.shift();
+    if (this.totalLenght > this.maxNodesToShow) {
+      this.collapsedElements = this.pathsList.splice(0, this.pathsList.length - 2);
+    } else {
+      this.collapsedElements = null;
+    }
+  }
+
+  componentDidRender(): void {
+    if (this.collapsedElementsRef) {
+      this.collapsedElementsRef.bindTo = this.triggerButton;
+      this.anchorElements = this.hostElement.querySelectorAll("z-list-group a");
+    }
+  }
+
+  private getPathsItemsList(): BreadcrumbPath[] | undefined {
+    return typeof this.paths === "string" ? JSON.parse(this.paths) : this.paths;
+  }
+
+  private renderMobileBreadcrumb(): HTMLDivElement {
+    const filteredPath = this.pathsList.filter((item) => !!item.path);
+    const lastPath = filteredPath[filteredPath.length - 1];
+
+    return (
+      <nav aria-label="Breadcrumb">
+        <ol class="mobile-breadcrumb">{this.renderNode(lastPath)}</ol>
+      </nav>
+    );
+  }
+
+  private renderHomepageNode(item): HTMLLIElement {
+    return (
+      <li>
+        <a
+          class="homepage"
+          href={item.path}
+          onClick={(e) => this.handlePreventFollowUrl(e, item)}
+        >
+          {this.homepageVariant === BreadcrumbHomepageVariant.ICON ? (
+            <z-icon
+              name="home"
+              fill="color-link-primary"
+              height={16}
+              width={16}
+            />
+          ) : (
+            "Home"
+          )}
+        </a>
+      </li>
+    );
+  }
+
+  private renderNode(item): HTMLLIElement {
+    return (
+      <li>
+        <a
+          aria-current={item.path ? undefined : "page"}
+          href={item.path}
+          onClick={(e) => this.handlePreventFollowUrl(e, item)}
+        >
+          {item.name}
+        </a>
+      </li>
+    );
+  }
+
+  private renderBreadcrumb(): HTMLElement {
+    return (
+      <nav aria-label="Breadcrumb">
+        <ol>
+          {this.renderHomepageNode(this.homepageNode)}
+          {this.collapsedElements ? this.renderOverflowMenu() : ""}
+          {this.pathsList.map((item) => this.renderNode(item))}
+        </ol>
+      </nav>
+    );
+  }
+
+  private togglePopover(): void {
+    if (!this.collapsedElementsRef.open) {
+      this.collapsedElementsRef.open = true;
+    }
+  }
+
+  private handlePreventFollowUrl(e: MouseEvent, item): void {
+    if (this.preventFollowUrl) {
+      e.preventDefault();
+      this.clickOnNode.emit(item.path);
+    }
+  }
+
+  private handleOverflowMenuAccessibility(e: KeyboardEvent): void {
+    if (e.key === KeyboardCode.TAB) {
+      e.preventDefault();
+
+      return;
+    }
+    e.stopPropagation();
+    const arrows = [KeyboardCode.ARROW_DOWN, KeyboardCode.ARROW_UP];
+    if (arrows.includes(e.key as KeyboardCode)) {
+      e.preventDefault();
+
+      if (e.key === KeyboardCode.ARROW_DOWN) {
+        this.currentIndex = [...this.anchorElements].length === this.currentIndex + 1 ? 0 : this.currentIndex + 1;
```

Comment:
potrebbe aver senso mettere [...this.anchorElements].length dentro una variabile e poi usare la variabile al posto che ripetere il codice ?